sycl/doc/extensions/proposed/sycl_ext_oneapi_syclbin.asciidoc

```diff
@@ -0,0 +1,239 @@
+  `bundle_state::input` and any device in `ctxt.get_devices()` does not have
+  `aspect::online_compiler`.
+* An `exception` with the `errc::invalid` error code if `State` is
+  `bundle_state::object` and any device in `ctxt.get_devices()` does not have
+  `aspect::online_linker`.
+* An `exception` with the `errc::build` error code if `State` is
+  `bundle_state::object` or `bundle_state::executable`, if the implementation
+  needs to perform an online compile or link, and if the online compile or link
+  fails.
+
+a|
+[frame=all,grid=none]
+!====
+a!
+[source]
+----
+namespace sycl::ext::oneapi::experimental {
+
+template<bundle_state State, typename PropertyListT = empty_properties_t>
+kernel_bundle<State> get_kernel_bundle(const context& ctxt,
+                                       const std::vector<device>& devs,
+                                       const std::filesystem::path& filename,
+                                       PropertyListT props = {}) (1)
+
+template<bundle_state State, typename PropertyListT = empty_properties_t>
+kernel_bundle<State> get_kernel_bundle(const context& ctxt,
+                                       const std::filesystem::path& filename,
+                                       PropertyListT props = {}) (2)
+
+}
+----
+!====
+
+_Effects (1):_ Creates a new kernel bundle containing the kernels inside the
+SYCLBIN file located at `filename`. This is equivalent to
+`get_kernel_bundle(ctxt, devs, data, props)` where `data` is the bytes read from
+the SYCLBIN file at `filename`.
+
+_Effects (2)_: Equivalent to `get_kernel_bundle(ctxt, ctxt.get_devices(), filename, props)`.
+
+_Constraints:_ Available only when `State` is not `ext_oneapi_source`.
+
+_Returns:_ The newly created kernel bundle, which has `State` state.
+
+_Throws:_
+
+* An `exception` with the `errc::invalid` error code if any of the devices in
+  `devs` is not one of devices contained by the context `ctxt` or is not a
+  descendent device of some device in `ctxt`.
+* An `exception` with the `errc::invalid` error code if the `devs` vector is
+  empty.
+* An `exception` with the `errc::invalid` error code if `State` is
+  `bundle_state::input` and any device in `ctxt.get_devices()` does not have
+  `aspect::online_compiler`.
+* An `exception` with the `errc::invalid` error code if `State` is
+  `bundle_state::object` and any device in `ctxt.get_devices()` does not have
+  `aspect::online_linker`.
+* An `exception` with the `errc::build` error code if `State` is
+  `bundle_state::object` or `bundle_state::executable`, if the implementation
+  needs to perform an online compile or link, and if the online compile or link
+  fails.
+
+|====
+
+```
+namespace sycl {
+template <bundle_state State> class kernel_bundle {
+public:
+  ...
+
+  std::vector<char> ext_oneapi_get_content();
+
+  sycl::span<char> ext_oneapi_get_content_sycl_view();
+
+#if __cplusplus >= 202002L
+  std::span<char> ext_oneapi_get_content_view();
```

Comment:
Since these are views, should they be `const char`?

Reply:
Yes.  We noticed the same thing in #16633.